source/support/spawn_vehicle.sqf

```diff
@@ -1,31 +1,34 @@
-_pos = _this select 0;
+#include "..\global_defines.h"
+
+_pos  = _this select 0;
 _name = _this select 1;
 _cost = _this select 2;
 
 _vehic = nil;
 
 _spawn_marker = {
-	_spawnPos = _this select 0;
-	_markername = str(format ["%1%2",_spawnPos select 0,_spawnPos select 1]);
-	_markerstr = createMarker [_markername, _spawnPos];
-	_markerstr setMarkerShape "ICON";
-	_markerstr setMarkerType "hd_dot";
-	_markerstr setMarkerColor "ColorBlue";
-	_markerstr setMarkerSize [1, 1];
-	_markerstr setMarkerText "vehicle";
-	sleep 60;
-	deletemarker _markerstr;
+    _spawnPos = _this select 0;
+    _markername = str(format ["%1%2",_spawnPos select 0,_spawnPos select 1]);
+    _markerstr = createMarker [_markername, _spawnPos];
+    _markerstr setMarkerShape "ICON";
+    _markerstr setMarkerType "hd_dot";
+    _markerstr setMarkerColor "ColorBlue";
+    _markerstr setMarkerSize [1, 1];
+    _markerstr setMarkerText "vehicle";
+    sleep 60;
+    deletemarker _markerstr;
 };
 
 if (commandpointsblu1 >= _cost) then {
-	hint "Vehicle ready! check map for location";
-	commandpointsblu1 = commandpointsblu1 - _cost;
-	ctrlSetText [1000, format["%1",commandpointsblu1]];
-	_vehic = _name createVehicle [0,0,0];
-	_vehic setPos (_pos);
-	//create a marker for the new vehicle
-	[_pos] spawn _spawn_marker;
+    hint "Vehicle ready! check map for location";
+    commandpointsblu1 = commandpointsblu1 - _cost;
+    ctrlSetText [MENU_COMMAND_POINTS_BLU, format["%1",commandpointsblu1]];
+    //spawning the vehicle in 0.0.0 then moving it is faster than spawning it in a specific location.
+    _vehic = _name createVehicle [0,0,0];
+    _vehic setPos (_pos);
+    //create a marker for the new vehicle
+    [_pos] spawn _spawn_marker;

```

Comment:
reminder, _spawn_marker code can be compiled (compileFinal) and added to global defines, this will be useful to add markers for debugging
required. add parameters and default values, compile and export.